core/src/main/java/com/sequenceiq/distrox/v1/distrox/StackUpgradeOperations.java

```diff
@@ -162,4 +181,21 @@ private boolean determineReplaceVmsParameter(Stack stack, Boolean replaceVms) {
             return upgradePreconditionService.notUsingEphemeralVolume(stack) && clusterDBValidationService.isGatewayRepairEnabled(stack.getCluster());
         }
     }
+
+    private List<String> getPreparedImagesList(long stackId) {
+        List<String> preparedImages = null;
```

Comment:
this is a bit antipattern to return with null collection. please return with an empty one instead

Reply:
done